modules/orakiBidAdapter.js

```diff
@@ -0,0 +1,217 @@
+    return valid;
+  },
+
+  buildRequests: (validBidRequests = [], bidderRequest = {}) => {
+    const device = deepAccess(bidderRequest, 'ortb2.device');
+    const page = deepAccess(bidderRequest, 'refererInfo.page', '');
+
+    let pageURL;
+    try {
+      pageURL = page && new URL(page);
+    } catch (e) {
+      logMessage(e);
+    }
+
+    const placements = [];
+    const request = {
+      deviceWidth: device?.w || 0,
+      deviceHeight: device?.h || 0,
+      language: device?.language?.split('-')[0] || '',
+      secure: pageURL.protocol === 'https:' ? 1 : 0,
+      host: pageURL.host,
+      page: pageURL.href,
+      placements,
+      coppa: deepAccess(bidderRequest, 'ortb2.regs.coppa') ? 1 : 0,
+      tmax: bidderRequest.timeout
+    };
+
+    if (bidderRequest.uspConsent) {
+      request.ccpa = bidderRequest.uspConsent;
+    }
+
+    if (bidderRequest.gdprConsent) {
+      request.gdpr = {
+        consentString: bidderRequest.gdprConsent.consentString
+      };
+    }
+
+    if (bidderRequest.gppConsent) {
+      request.gpp = bidderRequest.gppConsent.gppString;
+      request.gpp_sid = bidderRequest.gppConsent.applicableSections;
+    } else if (bidderRequest.ortb2?.regs?.gpp) {
+      request.gpp = bidderRequest.ortb2.regs.gpp;
+      request.gpp_sid = bidderRequest.ortb2.regs.gpp_sid;
+    }
+
+    const len = validBidRequests.length;
+    for (let i = 0; i < len; i++) {
+      const bid = validBidRequests[i];
+      placements.push(getPlacementReqData(bid));
+    }
+
+    return {
+      method: 'POST',
+      url: AD_URL,
+      data: request
+    };
+  },
+
+  interpretResponse: (serverResponse) => {
+    let response = [];
+    for (let i = 0; i < serverResponse.body.length; i++) {
+      let resItem = serverResponse.body[i];
+      if (isBidResponseValid(resItem)) {
+        const advertiserDomains = resItem.adomain && resItem.adomain.length ? resItem.adomain : [];
+        resItem.meta = { ...resItem.meta, advertiserDomains };
+
+        response.push(resItem);
+      }
+    }
+    return response;
+  },
+
+  getUserSyncs: (syncOptions, serverResponses, gdprConsent, uspConsent, gppConsent) => {
+    let syncType = syncOptions.iframeEnabled ? 'iframe' : 'image';
+    let syncUrl = SYNC_URL + `/${syncType}?pbjs=1`;
+
+    if (gdprConsent && gdprConsent.consentString) {
+      if (typeof gdprConsent.gdprApplies === 'boolean') {
+        syncUrl += `&gdpr=${Number(gdprConsent.gdprApplies)}&gdpr_consent=${gdprConsent.consentString}`;
+      } else {
+        syncUrl += `&gdpr=0&gdpr_consent=${gdprConsent.consentString}`;
+      }
+    }
+
+    if (uspConsent && uspConsent.consentString) {
+      syncUrl += `&ccpa_consent=${uspConsent.consentString}`;
+    }
+
+    if (gppConsent?.gppString && gppConsent?.applicableSections?.length) {
+      syncUrl += '&gpp=' + gppConsent.gppString;
+      syncUrl += '&gpp_sid=' + gppConsent.applicableSections.join(',');
+    }
+
+    const coppa = config.getConfig('coppa') ? 1 : 0;
```

Comment:
can you get coppa from the request on `bidderRequest.ortb2.regs` instead of from the config

Reply:
But this is `getUserSyncs`, there are no bidderRequest. Can you please explain how we can retrieve coppa in `getUserSyncs`? It will be greatly appretiated

Reply:
good call ty